edit-post/components/plugin-post-publish-panel/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Defines extensibility slot for post-publish sidebar.
+ *
+ * Defines slot to be used by plugins to insert content in the post-publish sidebar
+ * (which appears after a user fully publishes a post).
+ *
+ * @see /edit-post/README.md 
+ *
+ * @file   This files defines the PluginPostPublishPanel extension
+ * @author ChrisShultz
+ * @since  2.6.0
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Slot, Fill } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Name of slot to fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginPostPublishPanel';
+
+/**
+ * Renders the plugin sidebar fill.
+ *
+ * @return {WPElement} Plugin sidebar fill.
+ */
+function PluginPostPublishPanel( { children } ) {
```

Comment:
Should we introduce factory method to remove code duplication (similar to the new React Context AP)?:
```js
const createSlotFill = ( slotName ) => {
	function Component( { children } ) {
		return (
			<Fill name={ slotName } >
				{ children }
			</Fill>
		);
	}
	
	Component.displayName = slotName;

	Component.Slot = () => (
		<Slot name={ slotName } />
	);

	return Component;
};
```
then in this file:
```js
export default createSlotFill( 'PluginPostPublishPanel' );
```
/cc @aduth

Reply:
That's a good idea. No sense duplicating code. I can only assume there will be the need for more slots as Gutenberg evolves.

I wonder where `createSlotFill` should site. Potentially move it into a subfolder in the top level 'components' directory?

Reply:
Awesome! I'll look at incorporating this.